contracts/Collector.sol

```diff
@@ -56,46 +58,47 @@ contract Collector is ICollector {
 
     function updateShares(
         RevenuePartner[] memory partners
-    ) external validShares(partners) onlyOwner noBalanceToShare {
+    ) external onlyOwner noBalanceToShare updateValidShares(partners) {
         delete _partners;
-
-        for (uint256 i = 0; i < partners.length; i++)
-            _partners.push(partners[i]);
     }
 
     //@notice Withdraw the actual remainder and then update the remainder's address
     //for a new one. This function is the only way to withdraw the remainder.
     function updateRemainderAddress(
         address remainderAddress
     ) external onlyOwner noBalanceToShare {
-        uint256 balance = token.balanceOf(address(this));
-        address tokenAddr = address(token);
-
-        if (balance != 0) {
-            // solhint-disable-next-line avoid-low-level-calls
-            (bool success, bytes memory ret) = tokenAddr.call{gas: 200000}(
-                abi.encodeWithSelector(
-                    hex"a9059cbb",
-                    _remainderAddress,
-                    balance
-                )
-            );
-
-            require(
-                success && (ret.length == 0 || abi.decode(ret, (bool))),
-                "Unable to transfer remainder"
-            );
+        for (uint256 i = 0; i < _tokens.length; i++) {
+            IERC20 token = _tokens[i];
+            uint256 balance = token.balanceOf(address(this));
+
+            if (balance != 0) {
+                // solhint-disable-next-line avoid-low-level-calls
+                (bool success, bytes memory ret) = address(token).call(
+                    abi.encodeWithSelector(
+                        hex"a9059cbb",
+                        _remainderAddress,
+                        balance
+                    )
+                );
+
+                require(
+                    success && (ret.length == 0 || abi.decode(ret, (bool))),
+                    "Unable to transfer remainder"
+                );
+            }
         }
-
-        // solhint-disable-next-line
         _remainderAddress = remainderAddress;
     }
 
-    function getBalance() external view returns (uint256) {
-        return token.balanceOf(address(this));
+    function getTokens() external view returns (IERC20[] memory) {
+        return _tokens;
     }
 
-    function withdraw() external override onlyOwner {
+    function getRemainderAddress() external view returns (address) {
+        return _remainderAddress;
+    }
+
+    function withdrawToken(IERC20 token) public onlyOwner {
         uint256 balance = token.balanceOf(address(this));
```

Comment:
Would it be good idea to check if the `token` is in the `_tokens` array before withdrawing? Currently, `token` could be any ERC20 address. Since this function can only called by owner, I don't see much of an impact but still wanted to raise this issue.

Reply:
I am not convinced that it would be worth the extra computation/gas.

Reply:
@ulasanil Although it may make sense, I don't see any risk related to this, is there?

Reply:
It would just prevent unintended external calls to other smart contracts I guess. But I don't see a security risk. I think we can ignore this.

Reply:
Implemented